web3c/signer.js

```diff
@@ -0,0 +1,43 @@
+const nacl = require('tweetnacl');
+const hash = require('js-sha512').sha512_256;
+const bytes = require('./bytes');
+
+/**
+ * Given public key, a signer can verify web3c signatures over public keys from the public key.
+ */
+class Signer {
+
+  constructor(publicKey) {
+    this.publicKey = publicKey;
+  }
+
+  /**
+   * Ensures the signature is Sign_{key_manager}(hash(longTermKey || timestamp)).
+   *
+   * @param  {String} signature is a hex string signature of keyHex || timestamp.
+   * @param  {String} keyHex is a hex string of the long term public key.
+   * @param  {Number?} timestamp is the timestamp at which the key expires.
+   * @returns {Error} if signature !== Sign_{KEY_MANAGER_PUBLIC_KEY}(longTermKey || timestamp).
```

Comment:
```suggestion
   * @returns {Error} if signature !== Sign_{KEY_MANAGER_PUBLIC_KEY}(longTermKey || timestamp).
                       Otherwise returns null.
```